src/Analyzer/Resolve/IdentifierResolver.cpp

```diff
@@ -997,7 +1056,12 @@ IdentifierResolveResult IdentifierResolver::tryResolveIdentifierFromJoin(const I
         if (node->getNodeType() != QueryTreeNodeType::FUNCTION)
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected node type {}, expected function node", node->getNodeType());
 
-        const auto & function_argument_nodes = node->as<FunctionNode &>().getArguments().getNodes();
+        const auto & function_node = node->as<FunctionNode &>();
```

Comment:
TODO: maybe check if `check_nested_column_not_in_using` is needed at all.
I think the issue is only with the FULL JOIN. But if `firstTruthy` correctly deduces the type, subcolumns/nested should work (or throw an exception).

Also maybe now we can apply an optimization, like `getSubcolumn(firstTruthy(*), 'name') -> firstTruthy(getSubcolumn(*, 'name))`